[Composer] Update dependencies ahead of 2025.07 release

MrPetovan opened the pull request:
I have manually checked the diff for each of the main dependencies. My test node hasn't blown up either.

Main changes:
  - Upgrading asika/simple-console (1.0.3 => 1.0.4)
  - Upgrading composer/ca-bundle (1.5.0 => 1.5.8)
  - Upgrading dasprid/enum (1.0.5 => 1.0.7)
  - Upgrading ezyang/htmlpurifier (v4.17.0 => v4.18.0)
  - Upgrading guzzlehttp/guzzle (7.9.3 => 7.10.0)
  - Upgrading guzzlehttp/promises (2.2.0 => 2.3.0)
  - Upgrading guzzlehttp/psr7 (2.7.1 => 2.8.0)
  - Upgrading npm-asset/jquery-mousewheel (3.1.13 => 3.2.2)
  - Upgrading nyholm/psr7 (1.8.1 => 1.8.2)
  - Upgrading paragonie/certainty (v2.8.2 => v2.9.1)
  - Upgrading paragonie/constant_time_encoding (v2.6.3 => v2.8.2)
  - Upgrading paragonie/sodium_compat (v1.20.0 => v1.21.2)
  - Upgrading patrickschur/language-detection (v5.3.0 => v5.3.1)
  - Upgrading php-http/discovery (1.19.2 => 1.20.0)
  - Upgrading phpseclib/phpseclib (3.0.37 => 3.0.46)
  - Upgrading smarty/smarty (v4.5.3 => v4.5.6)
  - Upgrading symfony/polyfill-php80 (v1.31.0 => v1.33.0)
  - Upgrading ua-parser/uap-php (v3.9.14 => v3.10.0)

Dev changes:
  - Upgrading hamcrest/hamcrest-php (v2.0.1 => v2.1.1)
  - Upgrading mikey179/vfsstream (v1.6.11 => v1.6.12)
  - Upgrading mockery/mockery (1.6.10 => 1.6.12)
  - Upgrading myclabs/deep-copy (1.11.1 => 1.13.4)
  - Upgrading nikic/php-parser (v5.0.2 => v5.6.1)
  - Upgrading php-mock/php-mock (2.5.0 => 2.6.2)
  - Upgrading php-mock/php-mock-integration (2.3.0 => 3.0.0)
  - Upgrading php-mock/php-mock-phpunit (2.10.0 => 2.13.1)
  - Upgrading phpstan/phpstan (2.0.1 => 2.1.29)
  - Upgrading phpstan/phpstan-strict-rules (2.0.0 => 2.0.6)
  - Upgrading phpunit/php-code-coverage (9.2.31 => 9.2.32)
  - Upgrading phpunit/phpunit (9.6.17 => 9.6.29)
  - Upgrading sebastian/comparator (4.0.8 => 4.0.9)
  - Upgrading sebastian/exporter (4.0.6 => 4.0.8)
  - Upgrading sebastian/global-state (5.0.7 => 5.0.8)
  - Upgrading sebastian/recursion-context (4.0.5 => 4.0.6)
  - Upgrading symfony/polyfill-ctype (v1.31.0 => v1.33.0)
  - Upgrading symfony/polyfill-mbstring (v1.31.0 => v1.33.0)
  - Upgrading symfony/polyfill-php81 (v1.31.0 => v1.33.0)

MrPetovan committed:
[Composer] Update dependencies ahead of 2025.07 release

Main changes:
  - Upgrading asika/simple-console (1.0.3 => 1.0.4)
  - Upgrading composer/ca-bundle (1.5.0 => 1.5.8)
  - Upgrading dasprid/enum (1.0.5 => 1.0.7)
  - Upgrading ezyang/htmlpurifier (v4.17.0 => v4.18.0)
  - Upgrading guzzlehttp/guzzle (7.9.3 => 7.10.0)
  - Upgrading guzzlehttp/promises (2.2.0 => 2.3.0)
  - Upgrading guzzlehttp/psr7 (2.7.1 => 2.8.0)
  - Upgrading npm-asset/jquery-mousewheel (3.1.13 => 3.2.2)
  - Upgrading nyholm/psr7 (1.8.1 => 1.8.2)
  - Upgrading paragonie/certainty (v2.8.2 => v2.9.1)
  - Upgrading paragonie/constant_time_encoding (v2.6.3 => v2.8.2)
  - Upgrading paragonie/sodium_compat (v1.20.0 => v1.21.2)
  - Upgrading patrickschur/language-detection (v5.3.0 => v5.3.1)
  - Upgrading php-http/discovery (1.19.2 => 1.20.0)
  - Upgrading phpseclib/phpseclib (3.0.37 => 3.0.46)
  - Upgrading smarty/smarty (v4.5.3 => v4.5.6)
  - Upgrading symfony/polyfill-php80 (v1.31.0 => v1.33.0)
  - Upgrading ua-parser/uap-php (v3.9.14 => v3.10.0)

Dev changes:
  - Upgrading hamcrest/hamcrest-php (v2.0.1 => v2.1.1)
  - Upgrading mikey179/vfsstream (v1.6.11 => v1.6.12)
  - Upgrading mockery/mockery (1.6.10 => 1.6.12)
  - Upgrading myclabs/deep-copy (1.11.1 => 1.13.4)
  - Upgrading nikic/php-parser (v5.0.2 => v5.6.1)
  - Upgrading php-mock/php-mock (2.5.0 => 2.6.2)
  - Upgrading php-mock/php-mock-integration (2.3.0 => 3.0.0)
  - Upgrading php-mock/php-mock-phpunit (2.10.0 => 2.13.1)
  - Upgrading phpstan/phpstan (2.0.1 => 2.1.29)
  - Upgrading phpstan/phpstan-strict-rules (2.0.0 => 2.0.6)
  - Upgrading phpunit/php-code-coverage (9.2.31 => 9.2.32)
  - Upgrading phpunit/phpunit (9.6.17 => 9.6.29)
  - Upgrading sebastian/comparator (4.0.8 => 4.0.9)
  - Upgrading sebastian/exporter (4.0.6 => 4.0.8)
  - Upgrading sebastian/global-state (5.0.7 => 5.0.8)
  - Upgrading sebastian/recursion-context (4.0.5 => 4.0.6)
  - Upgrading symfony/polyfill-ctype (v1.31.0 => v1.33.0)
  - Upgrading symfony/polyfill-mbstring (v1.31.0 => v1.33.0)
  - Upgrading symfony/polyfill-php81 (v1.31.0 => v1.33.0)

annando commented:
The updated PHPStan now complains about some line in the addons that looks obscure but seems to be okay. Question is now we can fix it.

MrPetovan commented:
I fixed the error in `src/Content/NPF.php:332`, but I'm not sure what to do with the `phpmailer` code, it's using a variable variable which name is indeed empty by default, but it's never used as is. I'd have to look up the DKIM spec to understand why they used this feature.

MrPetovan commented:
Hmm, it's an external library, we probably should update it, but I don't have a good way to test the addon with a more recent version of PHPMailer.

MrPetovan committed:
Fix static analysis error in Content\Text\NPF

- Using explicit DOMAttr->name property instead of expecting the array key to be the attribute name

Art4 commented:
I suggest to let PHPStan exclude the complete `addon/mailstream/phpmailer/` folder in the `.phpstan-addons.neon` because it is a 3rd party lib. But nether the less we should also update PHPMailer or install it via composer.

MrPetovan committed:
Exclude deprecated PHPMailer library from PHPStan static analysis

- We should update PHPMailer or install it via Composer

MrPetovan commented:
I went for the former.

MrPetovan commented:
And we're back to the PHP 8.5 test errors.

Art4 reviewed (approved):
LGTM. We could ignore the PHP 8.5 error for now. I'll soon try to investigate the errors.
